modules/auxiliary/scanner/mssql/mssql_hashdump.rb

```diff
@@ -70,7 +70,7 @@ def run_host(ip)
     create_credential_login(login_data)
 
     # Grabs the Instance Name and Version of MSSQL(2k,2k5,2k8)
-    instancename= mssql_query(mssql_enumerate_servername())[:rows][0][0].split('\\')[1]
+    instancename= mssql_query(mssql_enumerate_servername())[:rows][0][0].split('\\')[0]
```

Comment:
Hm, it looks like this `[1]` was intentionally added previously:

https://github.com/rapid7/metasploit-framework/commit/a5e63c2ca995d85b6ec5e683f9f86718e706c428

I wonder if whatever your current string returns needed to be handled, as well as whatever value was previously expected?

Reply:
Interesting - IIRC the `[1]` was returning NULL at times even in master with `[0]` having the data we were looking for, I'll verify whether this is the right change or something I need to adjust elsewhere

Reply:
running this on both my branch and master, `mssql_query(mssql_enumerate_servername())[:rows][0][0].split('\\')` returns `["WIN-F0OBFMVGB07"]` - so the result of running this with `[1]` gives us `[*] 192.168.2.212:1433    - Instance Name: nil` on the output. Maybe I've got something configured wrong, but either way it seems like this should be tweaked to either [0] or there's some sort of logic we need to wrap this in

Reply:
I wonder if the behaviour should be preferencing second part, and if it's not there - fallback to the using the first part? 🤔 Worth a check :+1: